pkg/webhook/server.go

```diff
@@ -256,6 +256,7 @@ func (s *Server) mutate(receivedAdmissionReview v1beta1.AdmissionReview) *v1beta
 		AppendImagePullSecret: len(pod.Spec.ImagePullSecrets) > 0,
 		AppendInit:            len(pod.Spec.InitContainers) > 0,
 		AppendSidecar:         len(pod.Spec.Containers) > 0,
+		InjectFSGroup:         s.Config.EnableIAMForServiceAccounts && (pod.Spec.SecurityContext == nil || pod.Spec.SecurityContext.FSGroup == nil),
```

Comment:
should we automatically infer this if pod is marked with "fargate-profile"?

Reply:
@kiranmeduri it's not only for Fargate, this also applies for runs on EC2 node but using `iam-for-service-accounts`